src/main/java/com/hazelcast/kubernetes/HazelcastKubernetesDiscoveryStrategyFactory.java

```diff
@@ -34,6 +37,7 @@
  */
 public class HazelcastKubernetesDiscoveryStrategyFactory
         implements DiscoveryStrategyFactory {
+    private static final ILogger LOGGER = Logger.getLogger(HazelcastKubernetesDiscoveryStrategyFactory.class);
```

Comment:
nitpick: you could move this to the catch branch. as in most cases it wont be needed. and a static logger initialization has some disadvantages so we should avoid it unless necessary.